ariadne/contrib/sqlalchemy/query.py

```
@@ -0,0 +1,89 @@
+        object_types: Sequence[SQLAlchemyObjectType],
+    ):
+        super().__init__()
+        self.object_types = {ot.name: ot for ot in object_types}
+        self._object_types_by_model = {ot.model: ot for ot in object_types}
+
+    @staticmethod
+    def get_session_from_context(context: Any) -> Session | AsyncSession:
+        try:
+            return context["session"]
+        except KeyError:
+            raise RuntimeError("Session not found in context under key 'session'")
```

Ruff (0.15.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)